std/typecons.d

```diff
@@ -1801,7 +1810,15 @@ Forces $(D this) to the null state.
  */
     void nullify()()
     {
-        .destroy(_value);
+        static if (defaultConstructed)
+            .destroy(_value);
```

Comment:
.destroy will also destroy class *instance* (unfortunately), so this code as to be conditional. Adding `hasElaborateDestructor` is both enough to protect yourself from this case (classes don't have destructors, they have finalizers), as well as improve performance by not destroying things without destructors.

@JakobOvrum recently introduced `_destructRecurse`. @JakobOvrum: Does it make sense to call `_destructRecurse` directly here, or is `static if (hasElaborateDestructor!T) .destroy(_value)` the correct way to go?